### ISSUE TITLE
aws_elasticache_cluster doesn't allow availability_zone when replication_group_id is set

### ISSUE THREAD

**Issue by TheCloudlessSky** [terraform-providers/terraform-provider-aws#5579]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.32.0
+ provider.random v1.1.0

### Affected Resource(s)

* aws_elasticache_cluster

### Terraform Configuration Files

```hcl
resource "aws_elasticache_replication_group" "main_redis" {
  replication_group_id = "my-redis"
  engine = "redis"
  engine_version = "2.8.23"
  node_type = "cache.m3.medium"
  port = 6379
  subnet_group_name = "my-redis"
  security_group_ids = ["sg-..."]
  automatic_failover_enabled = true
  parameter_group_name = "default.redis2.8"
  at_rest_encryption_enabled = false
  transit_encryption_enabled = false
  maintenance_window = "sun:06:00-sun:07:00"
  auto_minor_version_upgrade = false
  apply_immediately = true
}

resource "aws_elasticache_cluster" "main_redis_001" {
  cluster_id = "my-redis-001"
  replication_group_id = "${aws_elasticache_replication_group.main_redis.id}"
  availability_zone = "us-east-1b"
  apply_immediately = true
}

resource "aws_elasticache_cluster" "main_redis_002" {
  cluster_id = "my-redis-002"
  replication_group_id = "${aws_elasticache_replication_group.main_redis.id}"
  availability_zone = "us-east-1a"
  apply_immediately = true
}
```

### Expected Behavior

Two Redis ElastiCache clusters should be created in availability zones `us-east-1a` and `us-east-1b` respectively. Even though `replication_group_id` is specified, you should still be able to customize the `availability_zone` of the cluster. The [AWS provider docs](https://www.terraform.io/docs/providers/aws/r/elasticache_replication_group.html#redis-cluster-mode-disabled) imply that the fine-grained control should be possible:

> Otherwise for fine grained control of the underlying cache clusters, they can be added or removed with the aws_elasticache_cluster resource and its replication_group_id attribute.

When you "Add Node" in the AWS Console, you can choose the availability zone. According to [the AWS docs for `CreateCacheCluster`](https://docs.aws.amazon.com/AmazonElastiCache/latest/APIReference/API_CreateCacheCluster.html):

> **ReplicationGroupId**
> 
> ...snip...
> 
> If the specified replication group is Multi-AZ enabled and the Availability Zone is not specified, the cluster is created in Availability Zones that provide the best spread of read replicas across Availability Zones.

Maybe I'm missing something but it seems like this should work. For example, setting it on `aws_elasticache_replication_group.availability_zones` seems wrong since [the docs say](https://www.terraform.io/docs/providers/aws/r/elasticache_replication_group.html#argument-reference):

> **availability_zones** - (Optional) A list of EC2 availability zones in which the replication group's cache clusters will be created. **The order of the availability zones in the list is not important.**

### Actual Behavior

I got an error (preventing the cache clusters from being created):

> Error: aws_elasticache_cluster.main_redis_001: "replication_group_id": conflicts with availability_zone
> Error: aws_elasticache_cluster.main_redis_002: "replication_group_id": conflicts with availability_zone

### Steps to Reproduce

1. `terraform validate` (or `plan` or `apply`)

### Important Factoids

None.

### References

None.

**Comment by bflad** [comment]
I was able to verify this is permitted via acceptance testing and submitted the fix: #5585

**Comment by TheCloudlessSky** [comment]
@bflad thanks! Just to confirm that I was also able to verify this with the AWS CLI.

Do you any ideas for a workaround until #5585 is merged/released? This is blocking us from using Terraform on a particular piece of our production infrastructure.

**Comment by bflad** [comment]
@TheCloudlessSky we generally don't commit to timelines for specific pull requests unless they're critical regression/crash fixes. In this case, the change was fairly trivial and was reviewed by another maintainer in time for the next release, so it will be released in version 1.33.0 of the AWS provider, likely sometime later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.33.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1330-august-22-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
